service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -764,6 +770,13 @@ public void editSiteEntry(JWTData userData, int entryId, UpdateSiteRequest editS
     siteEntriesRecord.setUserId(userData.getUserId());
     populateSiteEntry(siteEntriesRecord, editSiteEntryRequest);
 
-    siteEntriesRecord.store();
+    int siteId = siteEntriesRecord.getSiteId();
+
+    db.transaction(configuration -> {
+      siteEntriesRecord.store();
+      if (!editSiteEntryRequest.isTreePresent() && isAlreadyAdopted(siteId)) {
```

Comment:
Could we make sure that the entry that's being edited is the latest (i.e. most recent) site entry for a site here? If we're editing a site entry that's not the latest, we don't need to worry about unadopting (and we shouldn't force unadopt)